internal/controllers/clusterextension_controller_test.go

```diff
@@ -89,89 +89,17 @@ func TestClusterExtensionResolutionFails(t *testing.T) {
 	require.Equal(t, ocv1alpha1.ReasonFailed, cond.Reason)
 	require.Equal(t, fmt.Sprintf("no package %q found", pkgName), cond.Message)
 
-	verifyInvariants(ctx, t, reconciler.Client, clusterExtension)
-	require.NoError(t, cl.DeleteAllOf(ctx, &ocv1alpha1.ClusterExtension{}))
-}
-
-func TestClusterExtensionResolutionSucceeds(t *testing.T) {
```

Comment:
Removed this test function because I don't think we should be using a state we never expect to be returned as part of our standard code paths in a mock to end reconciliation without an error but still have successful resolution. Removing the unused, outside of testing, unpack states resulted in making the mock unpacker return an error. We already had a test (immediately after this one I think) that did this same logic so I renamed it and removed this test.

Reply:
Note: the github diff view makes this look a tad wonky